From 13ea2ac61f3dca36b6964913b6a7615953b5cc66 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 16 Nov 2020 17:36:27 +0200
Subject: [PATCH] [SW-12754] gaudi: security: prot boot strap pins

Since we no longer are toggling the PCIE_EN strap, we can keep it
protected and not exposed.
Protect mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS register.

Change-Id: Iefd190bc1e42dbe544626b0ebef19bddd7a5e4db
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index 8844336..2883c73 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -741,20 +741,10 @@
 		PROT_BIT(mmPSOC_GLOBAL_CONF_TRACE_ARUSER));
 	regmap_writel(mask, PRIV_PROT_REG(prot_base));
 
-	prot_base += SNGL_REG_PROT_SIZE;	/* offsets: 0x380 - 0x400 */
-	set_prot_bits_range_all(prot_base, SNGL_REG_PROT_SIZE);
+	prot_base += SNGL_REG_PROT_SIZE;	/* offsets: 0x380 - 0x480 */
+	set_prot_bits_range_all(prot_base, SNGL_REG_PROT_SIZE * 2);
 
-	/*
-	 * TODO: the register below is open temporarily to allow
-	 * LKD master to work correctly and condition the PCIe link init
-	 * properly.
-	 * mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS
-	 */
-	prot_base += SNGL_REG_PROT_SIZE;	/* offsets: 0x400 - 0x480 */
-	mask = ~PROT_BIT(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS);
-	regmap_writel(mask, PRIV_PROT_REG(prot_base));
-
-	prot_base += SNGL_REG_PROT_SIZE;	/* offsets: 0x480 - 0x500 */
+	prot_base += SNGL_REG_PROT_SIZE * 2;	/* offsets: 0x480 - 0x500 */
 	mask = ~PROT_BIT(mmPSOC_GLOBAL_CONF_SW_ALL_RST);
 	regmap_writel(mask, PRIV_PROT_REG(prot_base));
 
